### PR TITLE
feat: vod dynamic playback rate buffer control

### PR DESCRIPTION
**Pull request by gkatsev**
Dynamically update playback rate to keep the buffer full.

Related to https://github.com/shaka-project/shaka-player/issues/6131

### CODE REVIEW

**Review by avelad — APPROVED** [review]

File: externs/shaka/player.js [hunk 1]
```diff
@@ -1167,7 +1167,10 @@ shaka.extern.ManifestConfiguration;
  *   liveSyncPanicMode: boolean,
  *   liveSyncPanicThreshold: number,
  *   allowMediaSourceRecoveries: boolean,
- *   minTimeBetweenRecoveries: number
+ *   minTimeBetweenRecoveries: number,
+ *   vodDynamicPlaybackRate: boolean,
+ *   vodDynamicPlaybackRateMin: number,
+ *   vodDynamicPlaybackRateBufferPercentage: number
```
Comment (gkatsev) [inline externs/shaka/player.js:1173]: Naming is hard, so, if there are suggestions for better names, that would be appreciated.

Comment (avelad) [reply externs/shaka/player.js:1173]: @theodab can you review it? Thanks!

Comment (theodab) [reply externs/shaka/player.js:1173]: I don't like the name `vodDynamicPlaybackRateMin`. It suggests that this is the lower end of a range, and there is a `vodDynamicPlaybackRateMax`.
Perhaps if this is a 
I kind of want to say it should be called `vodDynamicPlaybackRateRate`, but that sounds silly. `vodDynamicPlaybackRateLowBufferRate` perhaps, even if it's a bit of a mouthful?

Also, `vodDynamicPlaybackRateBufferPercentage` being referred to as a "percentage" suggests it goes from 0 to 100, but it actually goes from 0 to 1. Perhaps `vodDynamicPlaybackRateBufferRatio`? Or just make it go from 0 to 100 and then divide by 100 inside the code.

Comment (gkatsev) [reply externs/shaka/player.js:1173]: > I kind of want to say it should be called `vodDynamicPlaybackRateRate`, but that sounds silly.

yeah, that's what I was thinking as well but then we need a separate name for the feature flag.

> `vodDynamicPlaybackRateLowBufferRate` perhaps, even if it's a bit of a mouthful?

seems reasonable for me. For a real mouthful see `liveSyncDynamicTargetLatencyStabilityThreshold` https://github.com/shaka-project/shaka-player/pull/6193/files#diff-9ac7cefa529d82513af9ff09186f9a6de19b063416e27c6fe97e834ea223388aR1175 (if you have thoughts there, would be helpful, though, Alvaro has thoughts on how to simplify the livesync section a bit).


> Perhaps `vodDynamicPlaybackRateBufferRatio`

sounds good.

Comment (gkatsev) [reply externs/shaka/player.js]: Upated the PR with the renaming.

**Comment by gkatsev** [comment]
This is ready for review again.

**Pull request merged**